plutus-contract/test/Spec/Contract/Ledger/Constraints/Spec.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{- | Integration tests covering each of the transaction constraints used by plutus-contract.
+    Currently covering both the plutus-ledger-constraints and plutus-tx-constraints. Soon only
+    plutus-tx-constraints will be supported.
+    These tests cover both the off chain and on chain constraints:
+      - Off chain constraints are used to build the transaction correctly to pass all phase-1
+        and phase-2 checks performed by the ledger.
+      - On chain constraints use plutus-tx to produce a script for validating all conditions.
+    The Trace Emulator is used to ensure the transaction constraints are being used similarly to
+    how developers building their applications would. This improves coverage of real-world scenarios
+    whilst also serving as a good reference for example usage.
```

Comment:
Nice top-level comment!

Do you think it would be useful to include the general testing strategy for the constraints? For example, the general pattern is that you would test the on-chain + off-chain implementation of _most_ constraints using a minting policy script because you would need a single transaction to test both behavior. Whereas, if you used a spending script, you would need two transactions in order to test both on-chain and off-chain behavior of a constraint.

Again, not sure how valuable it would be, but it did help me understand the general pattern of how you wrote the tests.

Reply:
I'll add to the comment to mention the reason for using minting policies. Benefits are mainly efficiency and readability.